--- unpatched_source/src/library/utils/R/zzz.R	2019-01-25 14:02:24.832369500 -0800
+++ source/src/library/utils/R/zzz.R	2019-01-25 14:04:20.388369500 -0800
@@ -38,8 +38,7 @@
         if(.Platform$OS.type == "windows") {
             list(unzip = "internal",
                  editor = if(length(grep("Rgui", commandArgs(), TRUE))) "internal" else "notepad",
-                 repos = c(CRAN = "@CRAN@",
-                           CRANextra = "http://www.stats.ox.ac.uk/pub/RWin"),
+                 repos = c(CRAN = "@CRAN@"),
                  askYesNo = if (.Platform$GUI == "Rgui") askYesNoWinDialog
                  )
         } else
